rtl/ibex_id_stage.sv

```
@@ -335,22 +334,14 @@ module ibex_id_stage #(
   // Registers //
   ///////////////
 
-  logic [31:0] regfile_wdata_mux;
-  logic        regfile_we_mux;
-  logic  [4:0] regfile_waddr_mux;
-
-  //TODO: add assertion
-  // Register File mux
-  always_comb begin
-    regfile_we_mux      = regfile_we;
-    regfile_waddr_mux   = regfile_alu_waddr_id;
-    if (select_data_rf == RF_LSU) begin
-      regfile_wdata_mux = regfile_wdata_lsu_i;
-    end else if (csr_access) begin
-      regfile_wdata_mux = csr_rdata_i;
-    end else begin
-      regfile_wdata_mux = regfile_wdata_ex_i;
-    end
+  // Register file write data mux
+  always_comb begin : regfile_wdata_mux
+    unique case (regfile_wdata_sel)
+      RF_WD_EX:  regfile_wdata_id = regfile_wdata_ex_i;
+      RF_WD_LSU: regfile_wdata_id = regfile_wdata_lsu_i;
+      RF_WD_CSR: regfile_wdata_id = csr_rdata_i;
+      default:   regfile_wdata_id = regfile_wdata_ex_i;
```

The default can't happen, can it? Can you remove the default statement, or output x here to avoid masking errors when extending the RF_WD_* enum in the future?

---

We can't remove the default (required by our guidelines) but we could output X. And I think it is a good idea to do so. I changed also the other muxes accordingly.